web/genericcom.js

```diff
@@ -47,6 +47,64 @@ class ExternalServices extends BaseExternalServices {
   }
 }
 
+class SignatureStorage {
+  // TODO: Encrypt the data in using a password and add a UI for entering it.
+  // We could use the Web Crypto API for this (see https://bradyjoslin.com/blog/encryption-webcrypto/
+  // for an example).
+
+  #signatures = null;
+
+  #save() {
+    localStorage.setItem("pdfjs.signature", JSON.stringify(this.#signatures));
+  }
+
+  async getAll() {
+    if (!this.#signatures) {
+      const data = localStorage.getItem("pdfjs.signature");
+      this.#signatures = data ? JSON.parse(data) : Object.create(null);
+    }
+    return this.#signatures;
+  }
+
+  async isFull() {
+    return Object.keys(await this.getAll()).length === 5;
+  }
+
+  async create(data) {
+    if (await this.isFull()) {
+      return null;
+    }
+    const uuid = crypto.randomUUID();
```

Comment:
This method is limited to secure context, and thus won't be available everywhere; see https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

Here, and elsewhere in this patch, please use our helper function instead:
https://github.com/mozilla/pdf.js/blob/e608daed9f25d22e35afad91190462ace06162f5/src/shared/util.js#L1076-L1086